### ISSUE TITLE
FP and FN due to treating `IsNullOrWhiteSpace` as is equal to null

### ISSUE THREAD

**Issue by abbas-sabra-sonarsource** [SonarSource/sonar-dotnet#2576]
### Description

"IsNullOrWhiteSpace" is treated as is null which cause false positive and false negative in the symbolic execution based rules.

### Expected behavior

```
string s= " ";`
 if (string.IsNullOrWhiteSpace(s1))  // TP S2589
 { 
   ... // always evaluated
 }
```
### Actual behavior

 ```
 string s= " ";
 if (string.IsNullOrWhiteSpace(s1)) { // FP S2583
  ... // this is considered dead code
 }
```
